src/displayapp/screens/Calculator.cpp

```diff
@@ -0,0 +1,399 @@
+      lv_obj_set_style_local_bg_color(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, Colors::bgAlt);
+      lv_obj_set_style_local_bg_grad_color(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, Colors::deepOrange);
+      lv_btnmatrix_set_btn_ctrl(buttonMatrix, 7, LV_BTNMATRIX_CTRL_CHECK_STATE);
+      break;
+    case '*':
+      lv_obj_set_style_local_bg_grad_dir(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, LV_GRAD_DIR_HOR);
+      lv_obj_set_style_local_bg_color(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, Colors::deepOrange);
+      lv_obj_set_style_local_bg_grad_color(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, Colors::bgAlt);
+      lv_btnmatrix_set_btn_ctrl(buttonMatrix, 11, LV_BTNMATRIX_CTRL_CHECK_STATE);
+      break;
+    case '/':
+      lv_obj_set_style_local_bg_grad_dir(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, LV_GRAD_DIR_HOR);
+      lv_obj_set_style_local_bg_color(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, Colors::bgAlt);
+      lv_obj_set_style_local_bg_grad_color(buttonMatrix, LV_BTNMATRIX_PART_BTN, LV_STATE_CHECKED, Colors::deepOrange);
+      lv_btnmatrix_set_btn_ctrl(buttonMatrix, 11, LV_BTNMATRIX_CTRL_CHECK_STATE);
+      break;
+    default:
+      lv_btnmatrix_clear_btn_ctrl_all(buttonMatrix, LV_BTNMATRIX_CTRL_CHECK_STATE);
+      break;
+  }
+}
+
+void Calculator::ResetInput() {
+  value = 0;
+  offset = FIXED_POINT_OFFSET;
+  operation = ' ';
+  equalSignPressed = false;
+  error = Error::None;
+}
+
+void Calculator::UpdateResultLabel() const {
+  int64_t integer = result / FIXED_POINT_OFFSET;
+  int64_t remainder = result % FIXED_POINT_OFFSET;
+  bool negative = (remainder < 0);
+
+  if (remainder == 0) {
+    lv_label_set_text_fmt(resultLabel, "%" PRId64, integer);
+    return;
+  }
+
+  if (remainder < 0) {
+    remainder = -remainder;
+  }
+
+  uint8_t min_width = N_DECIMALS;
+
+  // cut "0"-digits on the right
+  while ((remainder > 0) && (remainder % 10 == 0)) {
+    remainder /= 10;
+    min_width--;
+  }
+
+  if ((integer == 0) && negative) {
+    lv_label_set_text_fmt(resultLabel, "-0.%0*" PRId64, min_width, remainder);
+  } else {
+    lv_label_set_text_fmt(resultLabel, "%" PRId64 ".%0*" PRId64, integer, min_width, remainder);
+  }
+}
+
+void Calculator::UpdateValueLabel() {
+  switch (error) {
+    case Error::TooLarge:
+      lv_label_set_text_static(valueLabel, "too large");
+      break;
+    case Error::ZeroDivision:
+      lv_label_set_text_static(valueLabel, "zero division");
+      break;
+    case Error::None:
+    default: {
+      int64_t integer = value / FIXED_POINT_OFFSET;
+      int64_t remainder = value % FIXED_POINT_OFFSET;
+      bool negative = (remainder < 0);
+
+      int64_t printRemainder = remainder < 0 ? -remainder : remainder;
+
+      uint8_t min_width = 0;
+      int64_t tmp_offset = offset;
+
+      // TODO there has to be a simpler way to do this
```

Comment:
This should probably be resolved before merging

Reply:
Given that it works, seems correct, introduces no lag and I don't have an easier solution at hand, I would leave it like it is and remove the TODO.

Reply:
If the algorithm has turned out to be the best way to do it removing the todo sounds good to me too

Reply:
Nice. If you make a suggestion, I can commit it immediately.